### ISSUE TITLE
Responsive Switcher control as responsive condition

### ISSUE THREAD

**Issue by AmibeWebsites** [elementor/elementor#11618]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

I have two responsive controls. The first is a switcher control and the second has a condition of the switcher control having a value of 'yes'.

What I expected was that if the switcher is 'yes' for Desktop but not for Mobile that the second control would display for Desktop but not for Mobile, and vice versa.

What I experienced is that the condition only applies based on the Switchers value for Desktop.

## Steps to reproduce

Add a responsive Switcher control.
Add another responsive control conditional on the Switcher control.

In the editor, leave the Switcher off ('No') for Desktop but on ('Yes') for Mobile. The second responsive control remains hidden for both Desktop and Mobile. Change the Switcher to 'Yes' for Desktop and 'No' for mobile and the second responsive control is visible for both.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

**Comment by jahir07** [comment]
I found another solution :)

$this->add_responsive_control(
			'section_unique_id',
			[
				'label' => __( 'Show/Hide', 'elementor' ),
				'type' => Controls_Manager::SWITCHER,
				'label_on' => __( 'On', 'elementor' ),
				'label_off' => __( 'Off', 'elementor' ),
				'return_value'	=> 'none',
				'default'	=> 'block',
				'selectors' => [
					'{{WRAPPER}} .selectors' => 'display: {{VALUE}}',
				],
			]
		);

**Comment by yukasung** [comment]
**This** works for me.
`
$this->add_responsive_control(
            'show_excerpt',
            [
                'label' => esc_html__( 'Excerpt', 'elementor' ),
                'type' => Controls_Manager::SWITCHER,
                'label_on' => esc_html__( 'Show', 'elementor' ),
                'label_off' => esc_html__( 'Hide', 'elementor' ),
				'selectors_dictionary' => [
					'yes' => 'block',
					'' => 'none',
				],
                'default' => 'yes',
				'selectors' => [
					'{{WRAPPER}} . selectors' => 'display: {{VALUE}}',
				],
            ]
        );`

**Comment by ashotmark** [comment]
Hello Stephen, thanks for great question.

I faced exactly same problem and I found correct solution.

- First extends \Elementor\Widget_Common instead of Elementor\Widget_Base. 
-  \Elementor\Widget_Common will give you some useful method including get_responsive_device_args that will be used to specify control options as responsive.

so the sample code is looking like this:
`
$this->add_responsive_control('enable_ellipsis', [
            'type' => \Elementor\Controls_Manager::SWITCHER,
            'label' => esc_html__('Enable ellipsis', 'corelogic'),
            'label_on' => esc_html__('Yes', 'corelogic'),
            'label_off' => esc_html__('No', 'corelogic'),
            'return_value' => 'yes',
            'default' => 'no'
        ]);
        $this->add_responsive_control('ellipsis_rows', [
            'type' => \Elementor\Controls_Manager::SELECT,
            'label' => esc_html__('Ellipsis rows', 'corelogic'),
            'options' => [
                1 => 1,
                2 => 2,
                3 => 3,
                4 => 4,
                5 => 5,
                6 => 6
            ],
            'default' => 4,
            'condition' => [
                'enable_ellipsis' => 'yes'
            ],
            'device_args' => $this->get_responsive_device_args([
                'condition' => [
                    'enable_ellipsis_{{DEVICE}}' => ['yes'],
                ],
            ]),
        ]);
        `

please click follow button in my [github](https://github.com/ashotmark) if this helps you. thanks

**Comment by shilo-ey** [comment]
Hey all,

This issue has been resolved in Elementor v3.7.0

Feel free to update

Thanks!
